### PR TITLE
Category Module

### LINKED ISSUE #12: category module

### PR DESCRIPTION
**Pull request by botchway44**

### CODE REVIEW

File: src/category/category.service.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        private readonly categoryRepository: CategoryRepository
+    ) { }
+
+    async createCategory(
+        currentUser: User,
+        dto: CreateCategoryDTO) {
+        const category = new Category();
+
+        if (await this.categoryRepository.findByName(dto.name)) {
+            throw new ConflictException(MessageUtil.EXIST_CATEGORY_NAME);
```
Comment (KwabenBerko) [inline src/category/category.service.ts:21]: Hi Emmanuel, please follow the convention for the MessageUtil message when a resource already exists. It should be <RESOURCE>_ALREADY_ EXISTS.
So in this case CATEGORY_ALREADY_EXISTS

File: src/category/category.service.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        category.name = dto.name;
+        category.createdBy = currentUser;
+        return await this.categoryRepository.save(category);
+    }
+
+
+    async findByName(name: string) {
+        const category = await this.categoryRepository.findByName(name);
+        if (category === null) {
+            throw new NotFoundException(MessageUtil.CATEGORY_NOT_FOUND);
```
Comment (KwabenBerko) [inline src/category/category.service.ts:32]: Hi Emmanuel, please follow the convention for the MessageUtil message when a resource already exists. It should be **RESOURCE_ALREADY_ EXISTS**.
So in this case CATEGORY_ALREADY_EXISTS

File: src/category/category.module.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import { ObjectionModule } from "@willsoto/nestjs-objection";
+import { DatabaseModule } from "src/shared/database/database.module";
+import { CategoryRepository } from "./category.respository";
+import { CategoryService } from "./category.service";
+import { User } from "src/user/user.model";
+
+@Module({
+  imports: [
+    DatabaseModule,
+    ObjectionModule.forFeature([Category, User])
```
Comment (KwabenBerko) [inline src/category/category.module.ts]: Please take out the User model from the ObjectionModule.forFeature array and import the UserModule instead.

**Review by KwabenBerko — COMMENTED** [review]
Hi, please I've left some few comments.
Thank you.

File: src/category/category.service.ts (hunk 2 above)
Comment (botchway44) [reply src/category/category.service.ts:32]: Alright so fixed the RESOURCE_ALREADY_ EXISTS. But for RESOURCE_NOT_FOUND, think we should keep that

**Review by KwabenBerko — APPROVED** [review]

**Pull request merged**